socket_test.go

```diff
@@ -322,3 +322,68 @@ func TestConnMaxRetriesInfinite(t *testing.T) {
 		t.Fatalf("Dial called %d times, expected  at least %d", transport.dialCalledCount, atLeastExpectedRetries)
 	}
 }
+
+func TestSocketAutomaticReconnect(t *testing.T) {
+	listenEndpoint := "tcp://*:1234"
+	dialEndpoint := "tcp://localhost:1234"
```

Comment:
please use the `EndPoint("tcp")` function instead.
it will return an unused TCP port.